lib/spotify/spotify.go

```diff
@@ -24,9 +27,31 @@ func (s *Spotify) SetupRoutes(e *echo.Echo) {
 }
 
 func (s *Spotify) Authenticate() {
-	s.auth.authenticate()
-
 	go func() {
 		s.Client = s.auth.waitForClient()
 	}()
+
+	s.auth.authenticate()
+}
+
+// Don't question it: https://groups.google.com/g/golang-nuts/c/y9IvZgiNowk
+func callSpotify[R *Q, Q any](ctx context.Context, spot *Spotify, f func(ctx context.Context, opts ...spotifyLib.RequestOption) (R, error), opts ...spotifyLib.RequestOption) (R, error) {
+	var err error
+
+	if spot.Client == nil {
+		// TODO: return a custom error type
```

Comment:
Isn't this already done?
```suggestion
```

Reply:
It kind of was, but i meant an actual type, i've fixed it in the newest commit. Thank you for the reminder.